docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+ACK aims to provide the ability to “adopt” existing resources into your Kubernetes environment. Given the user has access to an existing resource, and that ACK operators exist for each resource’s associated service, the ACK controller will be able to determine the current specification and status of the AWS resource and reconcile said resource as if it were created natively in k8s.
+
+Users have requested this functionality as being pivotal to their adoption of ACK (see [#41](https://github.com/aws/aws-controllers-k8s/issues/41)). Information from user feedback has been considered as part of this document.
+
+Below are a number of use cases for requesting the ability this feature.
+
+1. Migrating existing architecture from another platform (CFN, CDK, Terraform, etc.)
+    1. Resources exist but have never been managed by K8s - no spec files
+2. Restoring from a broken K8s environment (etcd failure, complete state loss)
+    1. Resources exist and have previously been managed by K8s, but are not being actively reconciled
+3. Migrating architecture across K8s environments
+    1. Resource exist and are actively being managed by another K8s environment
+
+# Proposed Implementation
+
+## Overview
+
+The proposed solution will provide an alternate workflow by which to adopt resources through a separate set of manifests. A new custom resource definition, the `AdoptedResource` CRD, provides a standardised template by which to indicate to ACK that it should attempt to adopt a given resource. 
+The `AdoptedResource` CRD provides a boilerplate to reference an existing AWS resource (either through ARN, name or other unique identifier) and a target ACK type. Each ACK manager will subscribe to the list of adopted resources and will attempt to adopt any resources that match its managed type.
+The manager will describe the referenced AWS resource, typically through a `Describe*` SDK call, and use the returned values to compile a new custom resource - with filled spec and status fields. The manager will then apply this custom resource to the cluster using the metadata applied in the `AdoptedResource` spec.
+
+## `AdoptedResource` Specification
+
+The `AdoptedResource` custom resource definition will allow for specification of an AWS resource unique identifier, a target ACK type (with API group and kind), and target metadata fields.
+The AWS resource unique identifier allows identification through at most one of:
+
+* `arn` - An [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (Amazon Resource Name)
+* `name` - A resource name (a string value)
+* `id` - A resource ID (an integer value)
```

Comment:
I wonder if it would then be better to differentiate them, rather than by `name` and `id`, by `stringIdentifier` or `integerIdentifier` maybe? The reason I pointed out validation type was so each manager would already be provided with the cast type, rather than each of them having to parse to integer and give a corresponding error at runtime.

Reply:
What @vijtrip2 said :) The service controller will know how to look up a resource it is responsible for (either via a `name` field, an `id` field or an `arn` field). The AdoptedResource structure should not need to embed this controller-specific type-casting logic.